### PR TITLE
New implementation of dialog reading/sub dialog linking

### PR DESCRIPTION
**Pull request by dannyparsons**
New class: RCodeStructure which RFunction and ROperator inherit from and shifted some duplicate code into RCodeStructure
Added code to new user controls for nud, checkbox etc.
Started to get new dialog reading implemented on One Variable Graph dialog + sub dialog

### CODE REVIEW

File: instat/clsRLink.vb [hunk 1]
```diff
@@ -490,25 +490,25 @@ Public Class RLink
...
-                            If vecColumns.Count > 1 Then
-                                lstView.Items(j).Group = lstView.Groups(i)
-                            End If
-                        Next
-                    End If
-                Next
-                'TODO Find out how to get this to set automatically ( Width = -2 almost works)
-                lstView.Columns(0).Width = lstView.Width - 25
-            End If
+            For i = 0 To vecColumns.Count - 1
```
Comment (FrancoisJRenaud) [inline instat/clsRLink.vb]: Hi @dannyparsons  ! Is there anything important you changed here ? Seems like it is just spacing. clsRLink has been changed in the graph output pull request as well ...

Comment (dannyparsons) [inline instat/clsRLink.vb]: No I haven't. But it seems some spacing was not right so it auto corrected. Not sure if I can undo this.

**Comment by FrancoisJRenaud** [comment]
Are you going to rethink/rewrite RParameter ? I'ts sort of going the other way than clsRCodeStructure and it makes life complicated with the reader...
could contain an clsRCodeStructure instead of a couple of RFunction and ROperator ? Or would that not work ? Also is the way parameters are treated in RFunction and ROperator not close enough to have them in the clsRCodeStructure ? Maybe not... but I think it's worth trying to get overridable methods to add and manage parameters in the clsRCodeStructure ?
Syntactically, parameters inside an RFunction are just separated by an operator "," sort of. The only problem is the parameter name I guess... But then we also give parameter names to operator parameters anyway... So would just always be in "bIncludeParameterName = FALSE" mode.
 Also is this not the time to get things as an ordered list as parameters as sugested by David when thinking of reordering layers in the ucrLayerParameters...
?

**Comment by dannyparsons** [comment]
I'm adding this in now so that it doesn't cause conflicts. Still lots to do, but should be stable. @africanmathsinitiative/instat lots of changes to core structure, which I'll explain, but be on the look out for any new bugs that may have been introduced.

**Comment by FrancoisJRenaud** [comment]
@rdstern  suggested we should save a version of the software before merging the linking ....

**Pull request merged**